src/main/java/edu/cornell/kfs/sys/batch/MoveFileStep.java

```diff
@@ -0,0 +1,128 @@
+package edu.cornell.kfs.sys.batch;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.sys.batch.AbstractStep;
+
+import edu.cornell.kfs.sys.CUKFSConstants;
+
+/**
+ * Generic batch step for moving files from one directory to another.
+ * 
+ * At a minimum, the sourcePath and targetPath must be set to valid pre-existing directories,
+ * and the fileNamePattern must be set to a valid Java regex String. Only files whose names
+ * match the regex will be moved from the source directory to the target directory.
+ * This step will also automatically create ".done" files in the target directory for each moved file.
+ * 
+ * The source/target directory paths are not required to have trailing slashes. This step
+ * will automatically add them as needed.
+ */
+public class MoveFileStep extends AbstractStep {
+
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(MoveFileStep.class);
+
+    protected String sourcePath;
+    protected String targetPath;
+    protected Pattern fileNamePattern;
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        if (StringUtils.isBlank(sourcePath)) {
+            throw new IllegalStateException("sourcePath cannot be blank");
+        } else if (StringUtils.isBlank(targetPath)) {
+            throw new IllegalStateException("targetPath cannot be blank");
+        } else if (fileNamePattern == null) {
+            throw new IllegalStateException("fileNamePattern cannot be null");
+        }
+        
+        super.afterPropertiesSet();
+    }
+
+    @Override
+    public boolean execute(String jobName, Date jobRunDate) throws InterruptedException {
+        return moveFiles();
+    }
+
+    protected boolean moveFiles() {
+        verifyDirectoriesExist(sourcePath, targetPath);
+        
+        try {
+            File sourceDirectory = new File(sourcePath);
+            Matcher fileNameMatcher = fileNamePattern.matcher(StringUtils.EMPTY);
+            int numFilesMoved = 0;
+            
+            LOG.info("Moving files from " + sourcePath + " to " + targetPath);
+            
+            for (File sourceFile : FileUtils.listFiles(sourceDirectory, null, false)) {
+                fileNameMatcher.reset(sourceFile.getName());
+                if (fileNameMatcher.matches()) {
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug("Moving file: " + sourceFile.getName());

```

Comment:
I didn't know if we needed the file logging to be this fine-grained in a Production environment. I could certainly change it to INFO level if you think it would be more consistent with how things are currently set up in similar batch jobs.